xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -3642,12 +3642,33 @@ void IrEmitterUnnested::EmitFullWarpShuffleDownLoopForReduce(
   }
 }
 
-llvm::Value* IrEmitterUnnested::GetOutputAddressForReduction(
+static IrArray::Index TransformIndexForOutputFusion(
+    const IrArray::Index& hero_index, const HloReduceInstruction* hero,
+    const HloInstruction* root, llvm::IRBuilder<>* b) {
+  IrArray::Index out = hero_index;
+  const HloInstruction* it = root;
+  while (it != hero) {
```

Comment:
I think this can be simplified. Due to what kind of ops you allow as fusion epilogue, you can just immediately create SourceIndexOfBitcast from real hero to real root, without iterating the path.

Reply:
i think i was over-generalizing this logic for future extensions. will change it to keep it simple for now